pages/admin/view-templates.tsx

```diff
@@ -9,14 +9,23 @@ export const getServerSideProps = requireAuthentication(async (context) => {
     // getStaticProps is serverside, and therefore instead of doing a request,
     // we import the invoke Lambda function directly
 
-    const lambdaResult = await crudTemplates({ method: "GET" });
-    const templatesJSON =
-      lambdaResult &&
-      lambdaResult.data &&
-      lambdaResult.data.records &&
-      lambdaResult.data.records.length > 0
-        ? lambdaResult.data.records
-        : [];
+    const templatesJSON: unknown[] = [];
+
+    const getTemplatesFromLambda = async (limit = 13, offset = 0) => {
+      const lambdaResult = await crudTemplates({ method: "GET", limit: limit, offset: offset });
+
+      if (!lambdaResult || !lambdaResult.data || !lambdaResult.data.records) {
+        return;
+      }
+
+      Array.prototype.push.apply(templatesJSON, lambdaResult.data.records);
+
+      if (!(lambdaResult.data.records.length < limit)) {
```

Comment:
Inverse logic makes it difficult for readability. Easier as `if(lambdaResult.data.records.length >= limit)`

Reply:
I'm not sure I understand the logic here. If the number of records returned from the lambda is greater than the limit we recurse. So if 15 records are returned and the limit is 13 . The offset will be set to 13 on the first recursion meaning that we will have a duplication of two records. Furthermore shouldn't we only recurse if we get the number of records below the limit ?